cpp/src/arrow/util/parsing.h

```diff
@@ -119,8 +120,13 @@ class StringToFloatConverterMixin {
   }
 
  protected:
+// This is only support in double-conversion 3.1+
+#ifdef DOUBLE_CONVERSION_HAS_CASE_INSENSIBILITY
```

Comment:
Yuck. So I guess if double-conversion changes then Arrow have to be rebuilt

Reply:
No, `DOUBLE_CONVERSION_HAS_CASE_INSENSIBILITY` is a macro that is written into Arrow sources by CMake. The value of the macro won't change without an Arrow rebuild. Thus Arrow headers and Arrow binaries always match up.